src/firefly/js/ui/tap/ObjectIDSearch.jsx

```diff
@@ -50,11 +56,20 @@ export function ObjectIDSearch({cols, initArgs={}, capabilities, tableName,colum
         }
     );
 
+    //clickingSelectCols is toggled each time user clicks on the 'magnifying glass' to select cols, rendering this useEffect
+    useEffect(() => {
+        onTableLoaded(defaultTblId).then((tbl) => {
+            filterTable(cols,tbl);
+        });
+    }, [clickingSelectCols]);
+
```

Comment:
This doesn't feel like the cleanest way of doing this. But I needed a way to render this useEffect when the user clicks on the 'magnifying glass' to open the col selector popup. This is part of the `ColumnFld` component (in ColumnOrExpression.jsx) - so I needed to pass `clickingSelectCols` and its setter to this component (see line 265 of ObjectIDSearch.jsx). 

Now if you switch tables, or reopen the same column selector multiple times, it should retain the original filtering based on `meta.id` UCD entries (if multiple `meta.id` entries were not found, then no filtering is applied). 

Thoughts, @loitly @robyww?